libraries/AP_Airspeed/AP_Airspeed_ASP5033.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+  backend driver for airspeed from www.qio-tek.com it's a I2C ASP5033 sensor
+ */
+
+#include "AP_Airspeed_ASP5033.h"
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+
+extern const AP_HAL::HAL &hal;
+
+#define ASP5033_I2C_ADDR_1     0x6C
+#define ASP5033_I2C_ADDR_2     0x6D
+#define REG_SYS_CONFIG         0xA5
+#define REG_CMD                0x30
+#define REG_PRESS_DATA         0x06
+#define REG_TEMP_DATA          0x09
+#define REG_WAI                0x01
+#define REG_WAI_SET            0XA4
+#define REG_SENSOR_READY       0x08
+#define REG_WHOAMI_DEFAULT_ID  0X00
+#define REG_WHOAMI_RECHECK_ID  0X66
+#define CMD_MEASURE            0x0A
+#define RAW_DATA_ON            0b11111101
+
+
+
+bool AP_Airspeed_ASP5033::init()
+{
+    // probe the sensor
+    const uint8_t addresses[] = { ASP5033_I2C_ADDR_1, ASP5033_I2C_ADDR_2 };
+    for (uint8_t i=0; i<ARRAY_SIZE(addresses); i++) {
+        _dev = hal.i2c_mgr->get_device(get_bus(), addresses[i]);
+       if (!_dev && i <ARRAY_SIZE(addresses)) {
+         continue;
+       }
+       if(!_dev){
```

Comment:
This whole loop is broken, and will not handle the case where the device is never found. It needs to be reworked. It will also only ever detect the last I2C address not the first.